inference-engine/tests/functional/inference_engine/ir_serialization/transformation.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <fstream>
+
+#include "gtest/gtest.h"
+#include "ie_core.hpp"
+#include "ngraph/ngraph.hpp"
+#include "transformations/serialize.hpp"
+
+#ifndef IR_SERIALIZATION_MODELS_PATH  // should be already defined by cmake
+#define IR_SERIALIZATION_MODELS_PATH ""
+#endif
+
+class SerializationTransformationTest : public ::testing::Test {
+protected:
+    std::string test_name =
+        ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    std::string m_out_xml_path = test_name + ".xml";
+    std::string m_out_bin_path = test_name + ".bin";
+    std::shared_ptr<ngraph::Function> m_function;
+
+    void SetUp() override {
+        const std::string model = IR_SERIALIZATION_MODELS_PATH "add_abc.xml";
+        const std::string weights = IR_SERIALIZATION_MODELS_PATH "add_abc.bin";
+        InferenceEngine::Core ie;
+        m_function = ie.ReadNetwork(model, weights).getFunction();
+    }
+
+    void TearDown() override {
+        std::remove(m_out_xml_path.c_str());
+        std::remove(m_out_bin_path.c_str());
+    }
+};
+
+TEST_F(SerializationTransformationTest, DirectInstantiation) {
+    ngraph::pass::Serialize transform{m_out_xml_path, m_out_bin_path};
+    transform.run_on_function(m_function);
+
+    std::ifstream xml(m_out_xml_path);
+    std::ifstream bin(m_out_bin_path);
+    ASSERT_TRUE(xml.good());
+    ASSERT_TRUE(bin.good());
+}
+
+TEST_F(SerializationTransformationTest, PassManagerInstantiation) {
```

Comment:
Can we add a test to check that serialisation result is deterministic? (maybe I didn't find it).
Something like:
```
net->serialize(file1);
net->serialize(file2);
 no diff between file1 and file2
```

Reply:
Good idea, I'll add it. Can we add some library for hash calculation (e.g. md5) ? It would be the most straightforward solution - calculate file content hashes and compare them. If we can't add any library then i can compare it byte by byte. What do you think?

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/2940f9bf22c44d4aac8912096c52efddb4a99ac5). I went with binary comparison.